plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -356,9 +359,23 @@ func getPriorityFunctionConfigs(names sets.String, args PluginFactoryArgs) ([]al
 			})
 		}
 	}
+	var totalPriority int
+	for _, config := range configs {
+		// Checks for overflow and returns errors.
+		// TODO: Need to check overflow for weight*MaxPriority.
```

Comment:
Agree that we should handle overflow for `weight*MaxPriority` instead of `TODO`.

I'd suggest to move this check to `plugin/pkg/scheduler/api/validation/validation.go`.